hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -65,6 +79,12 @@ def __repr__(self):
         return f"integers({self.start}, {self.end})"
 
     def do_draw(self, data):
+        # For each finite endpoint, return that endpoint 2% of the time and one away 1% 
+        # of the time.
+        i = self.end_weighted_sampler.sample(data) if self.end_weighted_sampler else 0
+        if i < self.end_weighted_samples - 1:
+            return self.end_weighted_samples[i]
```

Comment:
Hey, @sclamons, I was taking a look at your changes (I'm also interested in the issue this tackles) but I can't understand this part of the code.

Notice how the `if` uses `self.end_weighted_samples` as if it were a number (because you are subtracting from it and comparing it to `i`) but in the return statement you index into it, as if `self.end_weighted_samples[i]` were a list or something of the sorts... 🤔